base/lnls-prune-artifacts.sh

```diff
@@ -77,7 +77,27 @@ get_used_epics_modules() {
     filter_out_paths "$all_modules" "$unused_modules"
 }
 
-remove_unused_epics_modules() {
+prune_module_dirs() {
+    module=$1
+
+    keep_paths="
+        $(find_shared_libs $module)
+        $(find $module -type f -regex ".*\.\(db\|template\|req\)" -printf "%h\n" | sort -u)
+    "
+
+    while read -r candidate; do
+        [ -d $candidate ] || continue
+
+        if [[ ! $keep_paths =~ "$candidate".* ]]; then
+            size=$(du -hs $candidate | cut -f 1)
+
+            printf "Removing directory '$candidate' ($size)...\n"
+            rm -rf $candidate
+        fi
```

Comment:
Is this nearly the same operation as `filter_out_paths`? Is it open-coded because you need to support spaces?

Reply:
Yes, and yes. However, regex matching suffers from what you said elsewhere: if a candidate folder is a prefix of a path to keep, this method decides to keep them. This means we might remove less than we could.

Maybe supporting file paths with spaces to `filter_out_paths` is a way to go. What do you think?

Reply:
That would mean you need to use arrays instead of lists in strings, right?

Reply:
Don't think so. `filter_out_paths` already receives two newline-separated strings. Stop using `for` and redirect the list to the `while read; do ... done` stdin should be enough.

Reply:
Doing so made possible to extract yet another function to be used both when removing entire modules and pruning only specific directories. ;)